solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -159,6 +166,10 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         // get expected target amount and fee
         (uint256 amount, uint256 fee) = targetAmountAndFee(_sourceToken, _targetToken, _amount);
 
+        // calculate protocol fee
+        uint256 calculatedProtocolFee = calculateProtocolFee(_targetToken, _amount);
+        amount = amount.sub(calculatedProtocolFee);

```

Comment:
protocol fee is applied to target token, but it uses `_amount` (which is denominated in the source tokens) as a base. Should be `amount` instead.